### DISCUSSION TITLE
Does semantic release support releasing daily/weekly instead of on every commit?

### DISCUSSION THREAD

**Discussion by HenrikPoulsen** [semantic-release/semantic-release#3070] [Q&A] [answered]
There's a lot of noisy packages out there that ship tons of releases every single day. So every time tools like Renovate and Dependabot etc runs there will always be a new release available.
I would prefer if the things I release are confined to a particular schedule, like releasing at most once per day etc.
Is this at all supported by semantic release? As in: would it properly detect all the things from the last release and put them in the new release with an appropriate version number?

The readme seems to imply that only publishing on every single commit to the release branch is supported.

**Comment by sheerlox** [comment, marked as answer]
While `semantic-release` was designed to automate the release process for Continuous Delivery purposes, it would also work perfectly if you ever decide to run the release workflow on a schedule :wink: 

On the other hand, because only the `fix` type triggers a release, that means your library/app should release a patch version _only if_ a production dependency was updated, which is best practice to make sure you're always integrating the latest bugfixes and/or security patches.

[By default, Renovate uses](https://docs.renovatebot.com/semantic-commits/) the `chore` type for development dependencies, and the `fix` type for actual dependencies that you ship to your consumers or production server, depending on whether you're building a library or an app.

**Reply by sheerlox** [reply]
@HenrikPoulsen I've updated my answer to mention it might be a good idea to keep running on every commit. Please let me know your thoughts and how your particular use case might differ.

**Reply by HenrikPoulsen** [reply]
In my specific case we have dependencies in our projects that essentially lead to new functionality being added/fixed/etc which is automerged (we have good test coverage).
I would not want to ship a new version potentially a hundred times per day just be cause a dependency does.
We could of course change the renovate schedule to only run weekly, but that would be a workaround to the problem.

 I want the code base to be kept up to date as much as possible and then we ensure we don't spam the consumers of it with several PRs per day.

We may end up just shipping per commit of course, as things tend to change when you have adopted a new work flow.
But for now I would like to keep the release schedule controlled and then we can work from there.

And if that's something semantic release can handle as well them I'm happy with that, since then I can at least start exploring it and see where we end up
